### PR TITLE
Add alpaka-ls

### PR DESCRIPTION
**Pull request by bernhardmgruber**
I needed to print the devices available for an accelerator, so I added a ~~test~~ small tool called `alpaka-ls` as example. ~~But maybe this should just be an example. What do you think?~~

### CODE REVIEW

**Comment by bernhardmgruber** [comment]
Thinking about this again, SYCL has a utility called `sycl-ls`, which for example prints this on my system:
```
bgruber@graviola:~$ sycl-ls
[opencl:acc:0] Intel(R) FPGA Emulation Platform for OpenCL(TM), Intel(R) FPGA Emulation Device 1.2 [2023.16.7.0.21_160000]
[opencl:cpu:1] Intel(R) OpenCL, AMD Ryzen 9 5950X 16-Core Processor             3.0 [2023.16.7.0.21_160000]
```
I think it could be useful to have an `alpaka-ls`, that lists similar information.

**Comment by fwyzard** [comment]
That's a good idea, and it could also be a good project for a new student.

**Comment by bernhardmgruber** [comment]
So, will this PR be merged or replaced by an example?

**Comment by SimeonEhrig** [comment]
Actual I would need the `alpaka-ls` tool for this issue: https://github.com/alpaka-group/alpaka/issues/2074

It was thinking about to implement CMake integration tests to test if our CMake is correctly working. Therefore I would implement a test, which builds `alpaka-ls` with several enabled backends. Then a bash script would execute it and check if the output matches with the enabled backends.

The background is, that I want to overwork the CMake totally, to support different correctly CMake targets and allows to enable all backends in parallel and compile a translation unit with a specific backend.

**Comment by bernhardmgruber** [comment]
Alright, here is the tool. Prints e.g. on my machine:
```
AccCpuSerial<1,int>
	AMD Ryzen 9 5950X 16-Core Processor
AccCpuOmp2Blocks<1,int>
	AMD Ryzen 9 5950X 16-Core Processor
AccGpuCudaRt<1,int>
	NVIDIA GeForce RTX 2060
```

**Comment by SimeonEhrig** [comment]
On HAL

```
AccCpuSerial<1,int>
	AMD EPYC 7452 32-Core Processor
AccCpuThreads<1,int>
	AMD EPYC 7452 32-Core Processor
AccGpuHipRt<1,int>
	AMD Radeon VII
	AMD Radeon VII
```

and

```
AccCpuSerial<1,int>
	AMD EPYC 7452 32-Core Processor
AccCpuThreads<1,int>
	AMD EPYC 7452 32-Core Processor
AccCpuOmp2Blocks<1,int>
	AMD EPYC 7452 32-Core Processor
AccCpuOmp2Threads<1,int>
	AMD EPYC 7452 32-Core Processor
AccGpuCudaRt<1,int>
	NVIDIA A30
```

Next week, also the Intel GPU is coming. So my goal after the renovation of the CMake is to display all 3 different GPUs with `alpaka-ls` :-)

File: example/ls/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        add_subdirectory("${CMAKE_CURRENT_LIST_DIR}/../.." "${CMAKE_BINARY_DIR}/alpaka")
+    else()
+        find_package(alpaka REQUIRED)
+    endif()
+endif()
+
+alpaka_add_executable(${PROJECT_NAME} src/ls.cpp)
+target_link_libraries(${PROJECT_NAME} PUBLIC alpaka::alpaka)
+set_target_properties(${PROJECT_NAME} PROPERTIES FOLDER example)
+add_test(NAME ${PROJECT_NAME} COMMAND ${CMAKE_PROJECT_NAME})
```
Comment (SimeonEhrig) [inline example/ls/CMakeLists.txt]:
```suggestion
add_test(NAME ${PROJECT_NAME} COMMAND ${PROJECT_NAME})
```

Fixes the broken test.

**Comment by bernhardmgruber** [comment]
@SimeonEhrig thx a lot for testing!

File: example/ls/CMakeLists.txt (hunk 1 above)
Comment (bernhardmgruber) [reply example/ls/CMakeLists.txt]: Applied

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**